docs/framework/react/guide/file-based-routing.md

```diff
@@ -91,6 +91,8 @@ The following options are available for configuration via the `tsr.config.json`
   - (Optional) If set, only route files and directories that start with this string will be considered for routing.
 - **`routeFileIgnorePrefix`**
   - (Optional, **Defaults to `-`**) Route files and directories that start with this string will be ignored. By default this is set to `-` to allow for the use of directories to house related files that do not contain any route files.
+- **`routeFileIgnoreType`**
+  - (Optional) ignore specific file formats in the route directory. It can be used in regular expression format. For example `.((css|const).ts)` can ignore `.css.ts` and `.const.ts` file
```

Comment:
this allows to ignore not only based on file extension, instead the whole filename can be used in the pattern

Reply:
right.

what I originally wanted was to simply exclude files that matched a specific file pattern, such as '.css.ts' or '.const.ts'. however, this code excludes all files and directories that match a certain regular expression, regardless of their type

although the initial setup may be more cumbersome and time-consuming to understand than using prefixes, it seems to have the advantage of being able to set more specific conditions for ignoring certain files and directories.

Reply:
https://github.com/TanStack/router/pull/1310/commits/81872bab5dd3f646c081efa9d34c393e3c0eec84

I changed description